src/protojson/JobErrorMsg.cc

```diff
@@ -0,0 +1,111 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "protojson/JobErrorMsg.h"
+
+#include <stdexcept>
+
+// Qserv headers
+#include "http/Client.h"
+#include "http/MetaModule.h"
+#include "http/RequestBodyJSON.h"
+#include "util/common.h"
+#include "util/TimeUtils.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+using namespace std;
+using namespace nlohmann;
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.protojson.JobErrorMsg");
+}  // namespace
+
+namespace lsst::qserv::protojson {
+
+JobErrorMsg::Ptr JobErrorMsg::create(std::string const& replicationInstanceId,
+                                     std::string const& replicationAuthKey, std::string const& workerIdStr,
+                                     std::string const& czarName, CzarIdType czarId, QueryId queryId,
+                                     UberJobId uberJobId, int errorCode, std::string const& errorMsg) {
+    auto jrMsg = Ptr(new JobErrorMsg(replicationInstanceId, replicationAuthKey));
```

Comment:
You may eliminate `std::`

Reply:
Youd need to add another specialized constructor to parse `json` and construct from `json` as I explained for class `JobReadyMsg`.